src/core/iomgr/tcp_windows.c

```diff
@@ -207,9 +219,12 @@ static void win_notify_on_read(grpc_endpoint *ep,
 
   /* Did we get data immediately ? Yay. */
   if (info->wsa_error != WSAEWOULDBLOCK) {
+    /* To avoid recursing forever, let's delay the callback for now. */
```

Comment:
This is a temporary workaround in the sense that the endpoint refactoring will take care of that. The code will be completely refactored away anyway in a greater, broader change that'll affect all platforms, not just Windows.